### PR TITLE
Add remaining indexes for foreign keys

### PR DESCRIPTION
**Pull request by Page-**
Also add explanation for some indexes that are for specific queries

Change-type: patch

### CODE REVIEW

File: src/migrations/00029-add-fk-indexes.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+CREATE INDEX CONCURRENTLY IF NOT EXISTS "device_is_running_release_idx"
+ON "device" ("is running-release");
+CREATE INDEX CONCURRENTLY IF NOT EXISTS "device_should_be_running_release_idx"
+ON "device" ("should be running-release");
+
+CREATE INDEX CONCURRENTLY IF NOT EXISTS "gateway_download_device_idx"
+ON "gateway download" ("is downloaded by-device");
+
+CREATE INDEX CONCURRENTLY IF NOT EXISTS "release_tag_release_idx"
+ON "release tag" ("release");
```
Comment (thgreasi) [inline src/migrations/00029-add-fk-indexes.sql:15]: I guess that this provides a measurable speed improvement on top of the composite release-tag_key unique index?

File: src/balena-init.sql [hunk 2]
```diff
@@ -12,86 +12,129 @@ ADD COLUMN IF NOT EXISTS "jwt secret" VARCHAR(255) NULL;
...
+-- "service environment variable"."service" is the first part of an automated unique index
+
+-- "service label"."service" is the first part of an automated unique index
+
+-- "service install"."device" is the first part of an automated unique index
+CREATE INDEX IF NOT EXISTS "service_install_service_idx"
+ON "service install" ("installs-service");
+
+CREATE INDEX IF NOT EXISTS "supervisor_release_device_type_idx"
+ON "supervisor release" ("is for-device type");
```
Comment (thgreasi) [inline src/balena-init.sql:110]: Any reason these two are not in the migration?
For sure they are not in my BoB's DB.

File: src/migrations/00029-add-fk-indexes.sql (hunk 1 above)
Comment (Page-) [reply src/migrations/00029-add-fk-indexes.sql]: Nope, this is a mistake - fixed it now

File: src/balena-init.sql (hunk 2 above)
Comment (Page-) [reply src/balena-init.sql:109]: I must have missed them when comparing what was moved vs new - fixed

**Comment by thgreasi** [comment]
Sad story:
I remember a few months ago while testing my orgs ui branch that I created a git branch with device_is_running_release_idx, device_should_be_running_release_idx which I later at some point deleted thinking that this was already merged...

**Pull request merged**